src/Parsers/examples/YAML_fuzzer.cpp

```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <time.h>
+#include <filesystem>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t * data, size_t size)
+{
+    /// How to test:
+    /// build ClickHouse with YAML_fuzzer.cpp
+    /// ./YAML_fuzzer YAML_CORPUS
+    /// where YAML_CORPUS is a directory with different YAML configs for libfuzzer
+
+    srand(time(NULL));
+    std::string cur_file = std::to_string(rand());
+
+    while (std::filesystem::exists(cur_file))
+    {
+        std::string cur_file = std::to_string(rand());
```

Comment:
I don't understand what's happening here. You generate a random number, then you transform it into a string, and after that you completely discard the resulting string. Have you tried to run this code?

Reply:
It generates random "filenames" and it does that until we find one which is not present in our directory, so we generate a new random number as long as file with this number exists. When we find a free number, we finally create a temporary file with that name.